Runtime/Model/MachineIdStorage.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Backtrace.Unity.Extensions;
+using System;
+using System.Linq;
+using System.Net.NetworkInformation;
+using UnityEngine;
+
+[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("Backtrace.Unity.Tests.Runtime")]
+namespace Backtrace.Unity.Model
+{
+    /// <summary>
+    /// Backtrace Machine Id storage
+    /// </summary>
+    internal class MachineIdStorage
+    {
+        /// <summary>
+        /// Player prefs machine identifier key
+        /// </summary>
+        internal const string MachineIdentifierKey = "backtrace-machine-id";
+
+        /// <summary>
+        /// Generate unique machine id. 
+        /// </summary>
+        /// <returns>Unique machine id Guid in a string format</returns>
+        internal string GenerateMachineId()
+        {
+            var storageMachineId = FetchMachineIdFromStorage();
+            if (!string.IsNullOrEmpty(storageMachineId))
+            {
+                return storageMachineId;
+            }
+#if !UNITY_WEBGL && !UNITY_SWITCH
+            var unityIdentifier = UseUnityIdentifier();
+            if (!GuidHelper.IsEmptyGuid(unityIdentifier))
+            {
+                StoreMachineId(unityIdentifier);
+                return unityIdentifier;
+            }
+            var networkIdentifier = UseNetworkingIdentifier();
+            if (!GuidHelper.IsEmptyGuid(networkIdentifier))
+            {
+                StoreMachineId(networkIdentifier);
+                return networkIdentifier;
+            }
+#endif
+            var backtraceRandomIdentifier = Guid.NewGuid().ToString();
+            StoreMachineId(backtraceRandomIdentifier);
+            return backtraceRandomIdentifier;
+        }
+
+
+        /// <summary>
+        /// Fetch a machine id in the internal storage
+        /// </summary>
+        /// <returns>machine identifier in the GUID string format</returns>
+        private string FetchMachineIdFromStorage()
+        {
+            return PlayerPrefs.GetString(MachineIdentifierKey);
+        }
+
+        /// <summary>
+        /// Set a machine id in the internal storage
+        /// </summary>
+        /// <param name="machineId">machine identifier</param>
+        private void StoreMachineId(string machineId)
+        {
+            PlayerPrefs.SetString(MachineIdentifierKey, machineId);
+        }
+
+        /// <summary>
+        /// Use Unity device identifier to generate machine identifier
+        /// </summary>
+        /// <returns>Unity machine identifier if the device identifier is supported. Otherwise null</returns>
+        protected virtual string UseUnityIdentifier()
+        {
+            // DeviceUniqueIdentifier will return "Switch" on Nintendo Switch
+            // try to generate random guid instead
```

Comment:
This comment doesn't seem to match what the code does

Reply:
Also, I think through the ifdef this code is never reached for Switch

Reply:
the flow makes sure this code won't be run against WebGL and Switch